### DISCUSSION TITLE
Correct task syntax?

### DISCUSSION THREAD

**Discussion by gogommartin** [cuelang/cue#404] [General] [answered]
Hi there, I'm using cue v0.2.0

I've seen at least 3 separate ways to write a task across the cue git repo

Option A as seen here [here](https://github.com/cuelang/cue/blob/v0.2.0/doc/tutorial/kubernetes/manual/services/dump_tool.cue)
```
command: hello: task: hello: {
  kind: "print"
  text: "hello"
}
```

Option B as seen [here](https://github.com/cuelang/cue/blob/v0.2.0/doc/tutorial/kubernetes/quick/services/dump_tool.cue)
```
command: hello: task: hello: cli.Print & {
  text: "hello"
}
```

Option C as seen [here](https://github.com/cuelang/cue/blob/v0.2.0/doc/cmd/cue.md)
```
command hello: task hello: cli.Print({
  text: "hello"
})
```

Option C throws an error when I use the example included in that documentation. Options A & B both work, but [this documentation](https://godoc.org/cuelang.org/go/pkg/tool) refers to the `kind` field as legacy with support for now. Based on this, and the output from `cue help cmd` I'm pretty confident that Option B is the correct syntax, but am looking for confirmation.

Thanks!

**Comment by mpvl** [comment, marked as answer]
Option C uses a lot of old and deprecated syntax. This documentation really should be generated.

The same holds for Option A. That should still work, but only as a grandfathered legacy style.

So you're right: Option B is the canonical way of writing things.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/404.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
